plugins/modules/locale_gen.py

```diff
@@ -33,11 +33,20 @@
   state:
     type: str
     description:
-      - Whether the locale shall be present.
+      - Whether the locales shall be present.
     choices: [absent, present]
     default: present
 notes:
-  - This module does not support RHEL-based systems.
+  - If C(/etc/locale.gen) exists, the module will assume to be using the B(glibc) mechanism, else if C(/var/lib/locales/supported.d/)
+    exists it will assume to be using the B(debian) mechanism, else it will raise an error.
+  - When using glibc mechanism, it will manage locales by editing C(/etc/locale.gen) and running C(locale-gen).
+  - When using debian mechanism, it will manage locales by editing C(/var/lib/locales/supported.d/local) and then running
+    C(locale-gen).
+  - Please note that the code path that uses debian mechanism has not been tested for a while, because Ubuntu is already using
```

Comment:
As written in https://github.com/ansible-collections/community.general/issues/9131#issuecomment-2563488068 I don't think this mechanism should be called "debian".